### PR TITLE
EWL-4283 refactor gulp

### PR DESCRIPTION
**Pull request by designerneil**
## Ticket(s)

Please do not submit a Pull Request without a relevant JIRA ticket or Github issue! If you are creating a new pattern or feature, create an issue describing the need for this feature in Github **first** and then link to it below.

**Jira Ticket**

- [EWL-4283: Refactor gulp](https://issues.ama-assn.org/browse/EWL-4283)

## Description

Updates our gulpfile to the latest gulp 4 standards.
This PR also removes unused tasks 
Refactors tasks to follow gulp 4 recommendations

## To Test

- [ ] First try making sure gulp serve works
- [ ] This should load a browser with patternlab
- [ ] Make sure all css and js loads correctly
- [ ] Makes changes to your local scss file and see if they are automagically updated in the browsers
- [ ] The same goes for js files
- [ ] Make sure there is a twig directory in the public folder when you do a gulp or gulp serve
- [ ] Try a gulp release. This should create a tag here https://github.com/AmericanMedicalAssociation/ama-style-guide-2/releases
- [ ] Make sure any tasks you were using before this modification are included

## Visual Regressions

Make sure all css and js libraries load correctly. 
Make sure everything is loaded correctly

## Relevant Screenshots/GIFs

n/a

## Remaining Tasks

none

## Additional Notes

This will have a global impact and needs to be tested thoroughly before merging. If all goes well gulp tasks will be speedier and less prone to breakage.

### CODE REVIEW

File: styleguide/package.json [hunk 1]
```diff
@@ -1,45 +1,44 @@
 {
   "name": "ama-style-guide",
-  "version": "1.13.5",
+  "version": "3.5.0",
```
Comment (labbydev) [inline styleguide/package.json]: This is tagging the release. What was the reasoning from going from `1.13.5` to `3.5.0`?

File: styleguide/package.json [hunk 2]
```diff
@@ -1,45 +1,44 @@
...
+    "gulp": "github:gulpjs/gulp#4.0",
+    "gulp-clean-css": "^3.9.2",
+    "gulp-concat": "^2.6.1",
+    "gulp-if": "^2.0.2",
     "gulp-plumber": "^1.1.0",
+    "gulp-rename": "^1.2.2",
     "gulp-sass-glob": "^1.0.8",
-    "gulp-svgmin": "^1.2.4",
-    "run-sequence": "^2.2.0"
+    "gulp-stylelint": "^6.0.0",
```
Comment (labbydev) [inline styleguide/package.json:37]: If in the future we want to have travis run a linter (we we probably will) this needs to be a dependency not a dev dependency.

File: styleguide/package.json [hunk 3]
```diff
@@ -1,45 +1,44 @@
...
     "patch-package": "^3.6.1",
     "postcss-reporter": "^5.0.0",
     "postcss-scss": "^1.0.2",
     "sass-grid": "^3.0.0"
   },
   "devDependencies": {
+    "argv": "0.0.2",
+    "del": "^3.0.0",
     "glob": "^7.1.2",
+    "gulp": "github:gulpjs/gulp#4.0",
```
Comment (labbydev) [inline styleguide/package.json:30]: This needs to be a full dependency so that Travis can run gulp as well.

File: styleguide/package.json [hunk 4]
```diff
@@ -1,45 +1,44 @@
...
+    "gulp-concat": "^2.6.1",
+    "gulp-if": "^2.0.2",
     "gulp-plumber": "^1.1.0",
+    "gulp-rename": "^1.2.2",
     "gulp-sass-glob": "^1.0.8",
-    "gulp-svgmin": "^1.2.4",
-    "run-sequence": "^2.2.0"
+    "gulp-stylelint": "^6.0.0",
+    "gulp-uglify": "^3.0.0",
+    "run-sequence": "^2.2.0",
```
Comment (labbydev) [inline styleguide/package.json:39]: This needs to be a full dependency so that Travis can run with run-sequence as well.

**Review by labbydev — COMMENTED** [review]
I'd like some more information on what tasks we're removing and why. Since the diff is too large I'd like to see everything outlined.

File: styleguide/package.json (hunk 4 above)
Comment (designerneil) [reply styleguide/package.json:39]: @labbydev gulp 4 does not use run sequence. It's already built in

Comment (designerneil) [reply styleguide/package.json:39]: gulp.series('travis', 'somethingElse', gulp.parallel('scripts', 'styles'),

**Comment by designerneil** [comment]
we'll probably have add the travis stuff in serial

**Pull request closed without merging**